############################################################################################
############################################################################################
############################### EWA MOA 778 with genetic ORM ###############################
############################################################################################
############################################################################################

# Run 24/05/21 - now updated to be run 080821 (WITHOUT eGFR)

# EWAS with MOA on 778 individuals, using a genetic ORM to adjust for relatedness
# Accounting for covariates in a fully adjusted model:
# age
# epismoker
# BMI
# Sex
# Wave
# Batch
# Depression status 
# 5x WBC imputed 

# Known pQTLs have also been regressed from the protein levels - residuals are used in 
# the models that therefore have known genetic effects (from Sun et al) adjusted for.

# This model will produce the fully adjusted EWAS results.

############################################################################################################

### Methylation file - 778

############################################################################################################

# Taken as per the fully adjusted model - unchanged between fully adjusted and pQTL runs.
# Only the protein levels have been altered (i.e. regressing known SNP effects from Sun et al).

# Commented out here for reference:

# ## Read in methylation file used for BayesR and transpose so CpGs are columns and IDs are rows
# library(data.table) 
# w2 = readRDS("/Cluster_Filespace/Marioni_Group/Rob/Somalogic/wave2-STRADL-mvals.rds")
# w3 = readRDS("/Cluster_Filespace/Marioni_Group/Rob/Somalogic/wave3-STRADL-mvals.rds")
# w2 = w2[which(row.names(w2) %in% row.names(w3)),]
# w3 = w3[which(row.names(w3) %in% row.names(w2)),]

# ## Combine w2 and w3 
# meth = cbind(w2,w3)

# ## Add in GS ids instead of DNAm_ids 
# stradl_ids <- read.table("/Cluster_Filespace/Marioni_Group/Rob/Somalogic/STRADL_DNAm_target_REM_17April2020.txt",header =T)
# id.778 <- read.csv("/Cluster_Filespace/Marioni_Group/Rob/Somalogic/IDs_778.csv")

# ## Subset to the people in the ID file 
# stradl_ids <- stradl_ids[which(stradl_ids$GS_id %in% id.778$x),]
# ids = stradl_ids$DNAm_id
# meth <- meth[,match(ids, colnames(meth))]
# table(stradl_ids$DNAm_id == colnames(meth))
# colnames(meth) <- stradl_ids$GS_id

# ## Prepare phenotypes file for covariates that we will use to pre-correct data 
# stradl_ids <- stradl_ids[,c("GS_id", "age_stradl", "sex", "wave", "Stradl_id")]

# # Add batch 
# w2_batch <- read.csv("/Cluster_Filespace/Marioni_Group/Rob/Somalogic/wave2_batch.csv") 
# w2_batch$Batch[w2_batch$Batch == 1]<- "w1_1"
# w2_batch$Batch[w2_batch$Batch == 2]<- "w1_2"
# w2_batch$Batch[w2_batch$Batch == 3]<- "w1_3"
# w2_batch$Batch[w2_batch$Batch == 4]<- "w1_4"
# w2_batch$Batch[w2_batch$Batch == 5]<- "w1_5"
# w2_batch$Batch[w2_batch$Batch == 6]<- "w1_6"
# w3_batch <- read.csv("/Cluster_Filespace/Marioni_Group/Rob/Somalogic/wave_3_original_samplesheet.csv") 
# w2_batch <- w2_batch[,c("Sample_Name","Batch")]
# names(w2_batch) <- c("GS_id", "Batch")
# w3_batch <- w3_batch[,c("Sample_Name","Batch_all")]
# names(w3_batch) <- c("GS_id", "Batch")
# w3_batch$GS_id <- gsub("ST", "", w3_batch$GS_id)
# batch = rbind(w2_batch, w3_batch)
# batch = batch[-which(duplicated(batch$GS_id)),]

# phenotypes= merge(stradl_ids, batch, by = "GS_id")

# # Add WBC info into phenotypes for regressions 
# w2cells <- read.csv("/Cluster_Filespace/Marioni_Group/Danni/Stradl_markers/EWAS_4000/stradl_w2_cells.csv")
# w3cells <- read.csv("/Cluster_Filespace/Marioni_Group/Danni/Stradl_markers/EWAS_4000/stradl_w3_cells.csv")
# merged <- rbind(w2cells, w3cells)
# names(merged)[1] <- "GS_id"
# WBC <- merged[c(1,3,7:12)]

# phenotypes = merge(phenotypes, WBC, by = "GS_id")

# # Add in BMI info into phenotypes for regressions 
# demo_table <- read.csv("/Cluster_Filespace/Marioni_Group/Danni/Stradl_markers/protAA/demo_table.csv")
# BMI <- demo_table[c(14,54)]

# phenotypes = merge(phenotypes, BMI, by = "GS_id")
# library(imputeTS)
# phenotypes$BMI = na_mean(phenotypes$BMI) # 3 individuals with no BMI assigned mean imputed 

# # Join in epismoker info for those in STRADL to linker file, then to cov file 
# epi <- readRDS("/Cluster_Filespace/Marioni_Group/Danni/stradl_epismoker.rds")
# names(epi)[1] <- "DNAm_id"

# phenotypes = merge(phenotypes, epi, by = "DNAm_id")

# # Add depression status (based on SCID and incident case diagnoses pre STRADL sampling), as cov 
# dep <- read.csv("/Cluster_Filespace/Marioni_Group/Danni/Stradl_markers/pheWAS/Depression_case_controls_200521/prep_files/SCID_joint_to_GP_hospital_combined.csv")
# dep <- dep[c(1,6)]
# dep$combined <- as.character(dep$combined)
# names(dep)[1] <- "Stradl_id"
# library(tidyverse)
# which(phenotypes$Stradl_id %in% dep$Stradl_id) # 2 people with no data, which we will need to assign 0 as they have no depression status of case recorded

# phenotypes = left_join(phenotypes, dep, by = "Stradl_id")
# phenotypes$combined[is.na(phenotypes$combined)] <- 0

# # Get the set of finalised phneotypes for regressions 
# phenotypes = phenotypes[c(2:5,7:16)]

# # Match ID order between phenotype and meth files 
# ids= colnames(meth)
# phenotypes = phenotypes[match(ids, as.character(phenotypes$GS_id)),]
# table(colnames(meth) == phenotypes$GS_id)

# # Write out covariates to use in heatmaps 
# # write.csv(phenotypes, "/Cluster_Filespace/Marioni_Group/Danni/Stradl_markers/pheWAS/corrplots/phenotypes_file_778_depression_added_240521.csv", row.names = F)

# # Transpose methylation data
# meth = t(meth)

# # Resdualise meth matrix 
# for(i in 1:(ncol(meth))){
#   print(i)
#   meth[,i] <- resid(lm(meth[,i] ~ phenotypes$age_stradl + phenotypes$smokingScore + phenotypes$BMI + as.factor(phenotypes$sex) + as.factor(phenotypes$Batch) + as.factor(phenotypes$wave) + as.factor(phenotypes$combined) + phenotypes$Bcell + phenotypes$CD4T + phenotypes$CD8T + phenotypes$Gran + phenotypes$NK, na.action = na.exclude))
# } 

# meth<-as.data.frame(meth)

# ## Combined IDs of individuals 
# id <- read.csv("/Cluster_Filespace/Marioni_Group/Rob/Somalogic/IDs_778.csv")
# fam <- read.table("/Cluster_Filespace/Marioni_Group/Rob/Somalogic/GWAS_Somalogic_data.fam")

# # Subset to 778
# fam1 = fam[fam$V2 %in% id$x,]
# ## Make sure order of IDs in Family ID and regular ID file match 
# matcher = id$x
# fam1 = fam1[match(matcher,fam1$V2),]
# table(fam1$V2 == id$x)

# # match meth file to the order in the 778_id file 
# ids = id$x
# meth <- meth[match(ids, row.names(meth)),]
# table(fam1$V2 == row.names(meth))

# ## Combine IDs for FID and IID and rearrange columns so it goes FID, IID, cpg1, cpg2... 
# tmp <- meth
# tmp$IID <- fam1$V2 
# tmp$FID <- fam1$V1 
# tmp <- tmp[,c(ncol(tmp), ncol(tmp)-1, 1:(ncol(tmp)-2))]
# tmp$FID <- as.character(tmp$FID)
# tmp$IID <- as.character(tmp$IID)

# # test3 <- tmp[c(1:3)]
# # identical(test3$IID, as.character(id$x)) # true 

# # Add GS_ to IDs 
# tmp$FID<-sub("^","GS_", tmp$FID)
# tmp$IID<-sub("^","GS_", tmp$IID)

# # This file has everything regressed 
# fwrite(tmp, file="/Cluster_Filespace/Marioni_Group/Danni/Stradl_markers/EWAS_4000/EWAS_080421/test_files_comparison_090421/methylation_778_meth_order_full_240521.txt", row.names=F, quote = F, sep=' ')

# # Save order of IDs from the methylation file 
# test3 <- tmp[c(1:3)]
# write.csv(test3, "/Cluster_Filespace/Marioni_Group/Danni/Stradl_markers/EWAS_4000/EWAS_080421/test_files_comparison_090421/record_of_IDS_for_methylation_778_meth_order_full_240521_v2.csv")

# # meth file basic order 
# o1 <- read.csv("/Cluster_Filespace/Marioni_Group/Danni/Stradl_markers/EWAS_4000/EWAS_080421/test_files_comparison_090421/record_of_IDS_for_methylation_778_meth_order_basic_240521_v2.csv")
# # meth file full order 
# o2 <- read.csv("/Cluster_Filespace/Marioni_Group/Danni/Stradl_markers/EWAS_4000/EWAS_080421/test_files_comparison_090421/record_of_IDS_for_methylation_778_meth_order_full_240521_v2.csv")
# # Check they match up in terms of IDs 
# identical(o1$X, o2$X)
# identical(o1$FID, o2$FID)
# identical(o1$IID, o2$IID)

############################################################################################################

### Phenotype file - 778

############################################################################################################

# Adapted from the fully adjusted model phenotype script, but with the pQTLs regressed phenotype file

# Read in order of meth files we are matching to 
order <- read.csv("/Cluster_Filespace/Marioni_Group/Danni/Stradl_markers/EWAS_4000/EWAS_080421/test_files_comparison_090421/record_of_IDS_for_methylation_778_meth_order_full_240521_v2.csv")
# order_old <- read.csv("/Cluster_Filespace/Marioni_Group/Danni/Stradl_markers/EWAS_4000/EWAS_080421/test_files_comparison_090421/record_of_IDS_for_methylation_778_meth_order.csv")

### REPLACE WITH NON eGFR FILE 
# pheno <- read.csv("/Cluster_Filespace/Marioni_Group/Danni/Stradl_markers/EWAS_4000/prep/Phenotype_file_778_eGFR_inlcuded_pQTLs_regressed.csv", check.names = F)
pheno2 <- read.csv("/Cluster_Filespace/Marioni_Group/Danni/Stradl_markers/EWAS_4000/prep/Phenotype_file_778_eGFR_removed_pQTLs_regressed.csv", check.names = F)
pheno2 <- pheno2[-2]

# Add FAM and ID columns 
id <- read.csv("/Cluster_Filespace/Marioni_Group/Rob/Somalogic/IDs_778.csv") # 778 people in EWAS
fam <- read.table("/Cluster_Filespace/Marioni_Group/Rob/Somalogic/GWAS_Somalogic_data.fam") # 1064 total LBC family IDs included, with other info 

## Make sure order of IDs in Family ID and regular ID file match 
# So the second column of the fam file is the participant ID, which we will make sure matches here 
fam1 = fam[fam$V2 %in% id$x,] # subset the fam file to the participant IDs which are in the ID file 
matcher = id$x # create order based on the id file 
fam1 = fam1[match(matcher,fam1$V2),] # match order of participant IDs between the 2 files (first column is fam ID)

# Check to make sure the id in the proteins file is the same as the id in the fam1 file 
identical(pheno2$id, fam1$V2) # TRUE

# Name properly 
osca_dat <- fam1[c(1,2)]
names(osca_dat) <- c("FID", "IID")

pheno2$IID <- osca_dat$IID
pheno2$FID <- osca_dat$FID

# Order so proteins are first 
pheno2 <- pheno2[c(2:4236,1,4237,4238)]

# Add GS_ to the IDs 
pheno2$FID<-sub("^","GS_", pheno2$FID)
pheno2$IID<-sub("^","GS_", pheno2$IID)

identical(pheno2$IID, order$IID)

# Make sure the order of the phenotype file is matched to the main meth file order 
pheno2 <- pheno2[match(order$IID, pheno2$IID),]

identical(pheno2$IID, order$IID)

# Write out phenotype files for a heritability run in the 4235 with eGFR included as covariate (2 locations - one with WBC and one without)

location <- "/Cluster_Filespace/Marioni_Group/Danni/Stradl_markers/EWAS_4000/EWAS_080821_full/batches/"

## Write out protein files so FID, IID, phenotype 
for(i in 1:350){ 
name <- names(pheno2)[i]
write.table(pheno2[,c("FID","IID",name)], file = paste0(location, "batch1/",name,".phen"), col.names = F, row.names=F, quote = F, sep=' ')
}

## Write out protein files so FID, IID, phenotype 
for(i in 351:700){ 
name <- names(pheno2)[i]
write.table(pheno2[,c("FID","IID",name)], file = paste0(location, "batch2/",name,".phen"), col.names = F, row.names=F, quote = F, sep=' ')
}

## Write out protein files so FID, IID, phenotype 
for(i in 701:1050){ 
name <- names(pheno2)[i]
write.table(pheno2[,c("FID","IID",name)], file = paste0(location, "batch3/",name,".phen"), col.names = F, row.names=F, quote = F, sep=' ')
}

## Write out protein files so FID, IID, phenotype 
for(i in 1051:1400){ 
name <- names(pheno2)[i]
write.table(pheno2[,c("FID","IID",name)], file = paste0(location, "batch4/",name,".phen"), col.names = F, row.names=F, quote = F, sep=' ')
}

## Write out protein files so FID, IID, phenotype 
for(i in 1401:1750){ 
name <- names(pheno2)[i]
write.table(pheno2[,c("FID","IID",name)], file = paste0(location, "batch5/",name,".phen"), col.names = F, row.names=F, quote = F, sep=' ')
}

## Write out protein files so FID, IID, phenotype 
for(i in 1751:2100){ 
name <- names(pheno2)[i]
write.table(pheno2[,c("FID","IID",name)], file = paste0(location, "batch6/",name,".phen"), col.names = F, row.names=F, quote = F, sep=' ')
}

## Write out protein files so FID, IID, phenotype 
for(i in 2101:2450){ 
name <- names(pheno2)[i]
write.table(pheno2[,c("FID","IID",name)], file = paste0(location, "batch7/",name,".phen"), col.names = F, row.names=F, quote = F,sep=' ')
}

## Write out protein files so FID, IID, phenotype 
for(i in 2451:2800){ 
name <- names(pheno2)[i]
write.table(pheno2[,c("FID","IID",name)], file = paste0(location, "batch8/",name,".phen"), col.names = F, row.names=F, quote = F, sep=' ')
}

## Write out protein files so FID, IID, phenotype 
for(i in 2801:3150){ 
name <- names(pheno2)[i]
write.table(pheno2[,c("FID","IID",name)], file = paste0(location, "batch9/",name,".phen"), col.names = F, row.names=F, quote = F,sep=' ')
}

## Write out protein files so FID, IID, phenotype 
for(i in 3151:3500){ 
name <- names(pheno2)[i]
write.table(pheno2[,c("FID","IID",name)], file = paste0(location, "batch10/",name,".phen"), col.names = F, row.names=F, quote = F, sep=' ')
}

## Write out protein files so FID, IID, phenotype 
for(i in 3501:3850){ 
name <- names(pheno2)[i]
write.table(pheno2[,c("FID","IID",name)], file = paste0(location, "batch11/",name,".phen"), col.names = F, row.names=F, quote = F,sep=' ')
}

## Write out protein files so FID, IID, phenotype 
for(i in 3851:4235){ 
name <- names(pheno2)[i]
write.table(pheno2[,c("FID","IID",name)], file = paste0(location, "batch12/",name,".phen"), col.names = F, row.names=F, quote = F,sep=' ')
}



# ############################################################################################################

# ### MAKE BINARY FILES AND ORM - 778 

# ############################################################################################################

# All preps follow the fully adjusted model script.


###########################################################################################

# BATCH EWAS 

cd /Cluster_Filespace/Marioni_Group/Danni/Stradl_markers/EWAS_4000/EWAS_080821_full/batches/batch1/
for i in *.phen 

do

  A=$( echo $i | cut -d"/" -f3)
  B=$( echo $A | cut -d. -f1)

osca_Linux --moa --orm /Cluster_Filespace/Marioni_Group/Danni/Stradl_markers/EWAS_4000/EWAS_080421/relatedness_testing_140421/daniel_orm/ormtest778 --befile /Cluster_Filespace/Marioni_Group/Danni/Stradl_markers/EWAS_4000/EWAS_080421/test_files_comparison_090421/methylation_778_meth_order_full_240521 --pheno $i --fast-linear --out ${B}_MOA_778_with_ORM --methylation-m


done 



cd /Cluster_Filespace/Marioni_Group/Danni/Stradl_markers/EWAS_4000/EWAS_080821_full/batches/batch2/
for i in *.phen 

do

  A=$( echo $i | cut -d"/" -f3)
  B=$( echo $A | cut -d. -f1)

osca_Linux --moa --orm /Cluster_Filespace/Marioni_Group/Danni/Stradl_markers/EWAS_4000/EWAS_080421/relatedness_testing_140421/daniel_orm/ormtest778 --befile /Cluster_Filespace/Marioni_Group/Danni/Stradl_markers/EWAS_4000/EWAS_080421/test_files_comparison_090421/methylation_778_meth_order_full_240521 --pheno $i --fast-linear --out ${B}_MOA_778_with_ORM --methylation-m


done 



cd /Cluster_Filespace/Marioni_Group/Danni/Stradl_markers/EWAS_4000/EWAS_080821_full/batches/batch3/
for i in *.phen 

do

  A=$( echo $i | cut -d"/" -f3)
  B=$( echo $A | cut -d. -f1)

osca_Linux --moa --orm /Cluster_Filespace/Marioni_Group/Danni/Stradl_markers/EWAS_4000/EWAS_080421/relatedness_testing_140421/daniel_orm/ormtest778 --befile /Cluster_Filespace/Marioni_Group/Danni/Stradl_markers/EWAS_4000/EWAS_080421/test_files_comparison_090421/methylation_778_meth_order_full_240521 --pheno $i --fast-linear --out ${B}_MOA_778_with_ORM --methylation-m


done 



cd /Cluster_Filespace/Marioni_Group/Danni/Stradl_markers/EWAS_4000/EWAS_080821_full/batches/batch4/
for i in *.phen 

do

  A=$( echo $i | cut -d"/" -f3)
  B=$( echo $A | cut -d. -f1)

osca_Linux --moa --orm /Cluster_Filespace/Marioni_Group/Danni/Stradl_markers/EWAS_4000/EWAS_080421/relatedness_testing_140421/daniel_orm/ormtest778 --befile /Cluster_Filespace/Marioni_Group/Danni/Stradl_markers/EWAS_4000/EWAS_080421/test_files_comparison_090421/methylation_778_meth_order_full_240521 --pheno $i --fast-linear --out ${B}_MOA_778_with_ORM --methylation-m


done 



cd /Cluster_Filespace/Marioni_Group/Danni/Stradl_markers/EWAS_4000/EWAS_080821_full/batches/batch5/
for i in *.phen 

do

  A=$( echo $i | cut -d"/" -f3)
  B=$( echo $A | cut -d. -f1)

osca_Linux --moa --orm /Cluster_Filespace/Marioni_Group/Danni/Stradl_markers/EWAS_4000/EWAS_080421/relatedness_testing_140421/daniel_orm/ormtest778 --befile /Cluster_Filespace/Marioni_Group/Danni/Stradl_markers/EWAS_4000/EWAS_080421/test_files_comparison_090421/methylation_778_meth_order_full_240521 --pheno $i --fast-linear --out ${B}_MOA_778_with_ORM --methylation-m


done 



cd /Cluster_Filespace/Marioni_Group/Danni/Stradl_markers/EWAS_4000/EWAS_080821_full/batches/batch6/
for i in *.phen 

do

  A=$( echo $i | cut -d"/" -f3)
  B=$( echo $A | cut -d. -f1)

osca_Linux --moa --orm /Cluster_Filespace/Marioni_Group/Danni/Stradl_markers/EWAS_4000/EWAS_080421/relatedness_testing_140421/daniel_orm/ormtest778 --befile /Cluster_Filespace/Marioni_Group/Danni/Stradl_markers/EWAS_4000/EWAS_080421/test_files_comparison_090421/methylation_778_meth_order_full_240521 --pheno $i --fast-linear --out ${B}_MOA_778_with_ORM --methylation-m


done 



cd /Cluster_Filespace/Marioni_Group/Danni/Stradl_markers/EWAS_4000/EWAS_080821_full/batches/batch7/
for i in *.phen 

do

  A=$( echo $i | cut -d"/" -f3)
  B=$( echo $A | cut -d. -f1)

osca_Linux --moa --orm /Cluster_Filespace/Marioni_Group/Danni/Stradl_markers/EWAS_4000/EWAS_080421/relatedness_testing_140421/daniel_orm/ormtest778 --befile /Cluster_Filespace/Marioni_Group/Danni/Stradl_markers/EWAS_4000/EWAS_080421/test_files_comparison_090421/methylation_778_meth_order_full_240521 --pheno $i --fast-linear --out ${B}_MOA_778_with_ORM --methylation-m


done 



cd /Cluster_Filespace/Marioni_Group/Danni/Stradl_markers/EWAS_4000/EWAS_080821_full/batches/batch8/
for i in *.phen 

do

  A=$( echo $i | cut -d"/" -f3)
  B=$( echo $A | cut -d. -f1)

osca_Linux --moa --orm /Cluster_Filespace/Marioni_Group/Danni/Stradl_markers/EWAS_4000/EWAS_080421/relatedness_testing_140421/daniel_orm/ormtest778 --befile /Cluster_Filespace/Marioni_Group/Danni/Stradl_markers/EWAS_4000/EWAS_080421/test_files_comparison_090421/methylation_778_meth_order_full_240521 --pheno $i --fast-linear --out ${B}_MOA_778_with_ORM --methylation-m


done 



cd /Cluster_Filespace/Marioni_Group/Danni/Stradl_markers/EWAS_4000/EWAS_080821_full/batches/batch9/
for i in *.phen 

do

  A=$( echo $i | cut -d"/" -f3)
  B=$( echo $A | cut -d. -f1)

osca_Linux --moa --orm /Cluster_Filespace/Marioni_Group/Danni/Stradl_markers/EWAS_4000/EWAS_080421/relatedness_testing_140421/daniel_orm/ormtest778 --befile /Cluster_Filespace/Marioni_Group/Danni/Stradl_markers/EWAS_4000/EWAS_080421/test_files_comparison_090421/methylation_778_meth_order_full_240521 --pheno $i --fast-linear --out ${B}_MOA_778_with_ORM --methylation-m


done 


cd /Cluster_Filespace/Marioni_Group/Danni/Stradl_markers/EWAS_4000/EWAS_080821_full/batches/batch10/
for i in *.phen 

do

  A=$( echo $i | cut -d"/" -f3)
  B=$( echo $A | cut -d. -f1)

osca_Linux --moa --orm /Cluster_Filespace/Marioni_Group/Danni/Stradl_markers/EWAS_4000/EWAS_080421/relatedness_testing_140421/daniel_orm/ormtest778 --befile /Cluster_Filespace/Marioni_Group/Danni/Stradl_markers/EWAS_4000/EWAS_080421/test_files_comparison_090421/methylation_778_meth_order_full_240521 --pheno $i --fast-linear --out ${B}_MOA_778_with_ORM --methylation-m


done 



cd /Cluster_Filespace/Marioni_Group/Danni/Stradl_markers/EWAS_4000/EWAS_080821_full/batches/batch11/
for i in *.phen 

do

  A=$( echo $i | cut -d"/" -f3)
  B=$( echo $A | cut -d. -f1)

osca_Linux --moa --orm /Cluster_Filespace/Marioni_Group/Danni/Stradl_markers/EWAS_4000/EWAS_080421/relatedness_testing_140421/daniel_orm/ormtest778 --befile /Cluster_Filespace/Marioni_Group/Danni/Stradl_markers/EWAS_4000/EWAS_080421/test_files_comparison_090421/methylation_778_meth_order_full_240521 --pheno $i --fast-linear --out ${B}_MOA_778_with_ORM --methylation-m


done 


cd /Cluster_Filespace/Marioni_Group/Danni/Stradl_markers/EWAS_4000/EWAS_080821_full/batches/batch12/
for i in *.phen 

do

  A=$( echo $i | cut -d"/" -f3)
  B=$( echo $A | cut -d. -f1)

osca_Linux --moa --orm /Cluster_Filespace/Marioni_Group/Danni/Stradl_markers/EWAS_4000/EWAS_080421/relatedness_testing_140421/daniel_orm/ormtest778 --befile /Cluster_Filespace/Marioni_Group/Danni/Stradl_markers/EWAS_4000/EWAS_080421/test_files_comparison_090421/methylation_778_meth_order_full_240521 --pheno $i --fast-linear --out ${B}_MOA_778_with_ORM --methylation-m


done 



####################################################################################################

### PROCESSING THE EWAS RESULTS FOR BATCHES ABOVE - MOA 778 with genetic ORM - AND covs - AND pQTLs

####################################################################################################

# Processing the 170521 results 

# Split this up for speed into 12 screens 

# Go through each protein and subset to the significant sites 

library(data.table)

list <- c("/Cluster_Filespace/Marioni_Group/Danni/Stradl_markers/EWAS_4000/EWAS_080821_full/batches/batch1/")

sig_hits <- "/Cluster_Filespace/Marioni_Group/Danni/Stradl_markers/EWAS_4000/EWAS_080821_full/batches/sig_hits/"

batch <- c("b1", "b2", "b3", "b4", "b5", "b6", "b7", "b8", "b9", "b10", "b11", "b12")

for(i in 1:length(list)){

  path <- list[i]
  batch <- batch[i]
  setwd(paste0(path))

  L <- list.files(".", ".mlma")

  # results tables for each protein (all vs sig)
  for(j in 1:length(L)){
    file <- fread(L[j], header = T)
    file <- as.data.frame(file)
    name <- as.character(L[j])
    name <- gsub("_.*", "", name)
    file$SeqId <- name
    write.csv(file, paste0(path, name, "_results_table.csv"), row.names = F)
    sub <- file[-which(file$p > 0.000000036),]
    write.csv(sub, paste0(path, name, "_results_sig.csv"), row.names = F)
    write.csv(sub, paste0(sig_hits, batch, "_", name, "_results_sig.csv"), row.names = F)
  }
}


library(data.table)

list <- c("/Cluster_Filespace/Marioni_Group/Danni/Stradl_markers/EWAS_4000/EWAS_080821_full/batches/batch2/")

sig_hits <- "/Cluster_Filespace/Marioni_Group/Danni/Stradl_markers/EWAS_4000/EWAS_080821_full/batches/sig_hits/"

batch <- c("b2")

for(i in 1:length(list)){

  path <- list[i]
  batch <- batch[i]
  setwd(paste0(path))

  L <- list.files(".", ".mlma")

  # results tables for each protein (all vs sig)
  for(j in 1:length(L)){
    file <- fread(L[j], header = T)
    file <- as.data.frame(file)
    name <- as.character(L[j])
    name <- gsub("_.*", "", name)
    file$SeqId <- name
    write.csv(file, paste0(path, name, "_results_table.csv"), row.names = F)
    sub <- file[-which(file$p > 0.000000036),]
    write.csv(sub, paste0(path, name, "_results_sig.csv"), row.names = F)
    write.csv(sub, paste0(sig_hits, batch, "_", name, "_results_sig.csv"), row.names = F)
  }
}


library(data.table)

list <- c("/Cluster_Filespace/Marioni_Group/Danni/Stradl_markers/EWAS_4000/EWAS_080821_full/batches/batch3/")

sig_hits <- "/Cluster_Filespace/Marioni_Group/Danni/Stradl_markers/EWAS_4000/EWAS_080821_full/batches/sig_hits/"

batch <- c("b3")

for(i in 1:length(list)){

  path <- list[i]
  batch <- batch[i]
  setwd(paste0(path))

  L <- list.files(".", ".mlma")

  # results tables for each protein (all vs sig)
  for(j in 1:length(L)){
    file <- fread(L[j], header = T)
    file <- as.data.frame(file)
    name <- as.character(L[j])
    name <- gsub("_.*", "", name)
    file$SeqId <- name
    write.csv(file, paste0(path, name, "_results_table.csv"), row.names = F)
    sub <- file[-which(file$p > 0.000000036),]
    write.csv(sub, paste0(path, name, "_results_sig.csv"), row.names = F)
    write.csv(sub, paste0(sig_hits, batch, "_", name, "_results_sig.csv"), row.names = F)
  }
}



library(data.table)

list <- c("/Cluster_Filespace/Marioni_Group/Danni/Stradl_markers/EWAS_4000/EWAS_080821_full/batches/batch4/")

sig_hits <- "/Cluster_Filespace/Marioni_Group/Danni/Stradl_markers/EWAS_4000/EWAS_080821_full/batches/sig_hits/"

batch <- c("b4")

for(i in 1:length(list)){

  path <- list[i]
  batch <- batch[i]
  setwd(paste0(path))

  L <- list.files(".", ".mlma")

  # results tables for each protein (all vs sig)
  for(j in 1:length(L)){
    file <- fread(L[j], header = T)
    file <- as.data.frame(file)
    name <- as.character(L[j])
    name <- gsub("_.*", "", name)
    file$SeqId <- name
    write.csv(file, paste0(path, name, "_results_table.csv"), row.names = F)
    sub <- file[-which(file$p > 0.000000036),]
    write.csv(sub, paste0(path, name, "_results_sig.csv"), row.names = F)
    write.csv(sub, paste0(sig_hits, batch, "_", name, "_results_sig.csv"), row.names = F)
  }
}



library(data.table)

list <- c("/Cluster_Filespace/Marioni_Group/Danni/Stradl_markers/EWAS_4000/EWAS_080821_full/batches/batch5/")

sig_hits <- "/Cluster_Filespace/Marioni_Group/Danni/Stradl_markers/EWAS_4000/EWAS_080821_full/batches/sig_hits/"

batch <- c("b5")

for(i in 1:length(list)){

  path <- list[i]
  batch <- batch[i]
  setwd(paste0(path))

  L <- list.files(".", ".mlma")

  # results tables for each protein (all vs sig)
  for(j in 1:length(L)){
    file <- fread(L[j], header = T)
    file <- as.data.frame(file)
    name <- as.character(L[j])
    name <- gsub("_.*", "", name)
    file$SeqId <- name
    write.csv(file, paste0(path, name, "_results_table.csv"), row.names = F)
    sub <- file[-which(file$p > 0.000000036),]
    write.csv(sub, paste0(path, name, "_results_sig.csv"), row.names = F)
    write.csv(sub, paste0(sig_hits, batch, "_", name, "_results_sig.csv"), row.names = F)
  }
}


library(data.table)

list <- c("/Cluster_Filespace/Marioni_Group/Danni/Stradl_markers/EWAS_4000/EWAS_080821_full/batches/batch6/")

sig_hits <- "/Cluster_Filespace/Marioni_Group/Danni/Stradl_markers/EWAS_4000/EWAS_080821_full/batches/sig_hits/"

batch <- c("b6")

for(i in 1:length(list)){

  path <- list[i]
  batch <- batch[i]
  setwd(paste0(path))

  L <- list.files(".", ".mlma")

  # results tables for each protein (all vs sig)
  for(j in 1:length(L)){
    file <- fread(L[j], header = T)
    file <- as.data.frame(file)
    name <- as.character(L[j])
    name <- gsub("_.*", "", name)
    file$SeqId <- name
    write.csv(file, paste0(path, name, "_results_table.csv"), row.names = F)
    sub <- file[-which(file$p > 0.000000036),]
    write.csv(sub, paste0(path, name, "_results_sig.csv"), row.names = F)
    write.csv(sub, paste0(sig_hits, batch, "_", name, "_results_sig.csv"), row.names = F)
  }
}



library(data.table)

list <- c("/Cluster_Filespace/Marioni_Group/Danni/Stradl_markers/EWAS_4000/EWAS_080821_full/batches/batch7/")

sig_hits <- "/Cluster_Filespace/Marioni_Group/Danni/Stradl_markers/EWAS_4000/EWAS_080821_full/batches/sig_hits/"

batch <- c("b7")

for(i in 1:length(list)){

  path <- list[i]
  batch <- batch[i]
  setwd(paste0(path))

  L <- list.files(".", ".mlma")

  # results tables for each protein (all vs sig)
  for(j in 1:length(L)){
    file <- fread(L[j], header = T)
    file <- as.data.frame(file)
    name <- as.character(L[j])
    name <- gsub("_.*", "", name)
    file$SeqId <- name
    write.csv(file, paste0(path, name, "_results_table.csv"), row.names = F)
    sub <- file[-which(file$p > 0.000000036),]
    write.csv(sub, paste0(path, name, "_results_sig.csv"), row.names = F)
    write.csv(sub, paste0(sig_hits, batch, "_", name, "_results_sig.csv"), row.names = F)
  }
}


library(data.table)

list <- c("/Cluster_Filespace/Marioni_Group/Danni/Stradl_markers/EWAS_4000/EWAS_080821_full/batches/batch8/")

sig_hits <- "/Cluster_Filespace/Marioni_Group/Danni/Stradl_markers/EWAS_4000/EWAS_080821_full/batches/sig_hits/"

batch <- c("b8")

for(i in 1:length(list)){

  path <- list[i]
  batch <- batch[i]
  setwd(paste0(path))

  L <- list.files(".", ".mlma")

  # results tables for each protein (all vs sig)
  for(j in 1:length(L)){
    file <- fread(L[j], header = T)
    file <- as.data.frame(file)
    name <- as.character(L[j])
    name <- gsub("_.*", "", name)
    file$SeqId <- name
    write.csv(file, paste0(path, name, "_results_table.csv"), row.names = F)
    sub <- file[-which(file$p > 0.000000036),]
    write.csv(sub, paste0(path, name, "_results_sig.csv"), row.names = F)
    write.csv(sub, paste0(sig_hits, batch, "_", name, "_results_sig.csv"), row.names = F)
  }
}



library(data.table)

list <- c("/Cluster_Filespace/Marioni_Group/Danni/Stradl_markers/EWAS_4000/EWAS_080821_full/batches/batch9/")

sig_hits <- "/Cluster_Filespace/Marioni_Group/Danni/Stradl_markers/EWAS_4000/EWAS_080821_full/batches/sig_hits/"

batch <- c("b9")

for(i in 1:length(list)){

  path <- list[i]
  batch <- batch[i]
  setwd(paste0(path))

  L <- list.files(".", ".mlma")

  # results tables for each protein (all vs sig)
  for(j in 1:length(L)){
    file <- fread(L[j], header = T)
    file <- as.data.frame(file)
    name <- as.character(L[j])
    name <- gsub("_.*", "", name)
    file$SeqId <- name
    write.csv(file, paste0(path, name, "_results_table.csv"), row.names = F)
    sub <- file[-which(file$p > 0.000000036),]
    write.csv(sub, paste0(path, name, "_results_sig.csv"), row.names = F)
    write.csv(sub, paste0(sig_hits, batch, "_", name, "_results_sig.csv"), row.names = F)
  }
}



library(data.table)

list <- c("/Cluster_Filespace/Marioni_Group/Danni/Stradl_markers/EWAS_4000/EWAS_080821_full/batches/batch10/")

sig_hits <- "/Cluster_Filespace/Marioni_Group/Danni/Stradl_markers/EWAS_4000/EWAS_080821_full/batches/sig_hits/"

batch <- c("b10")

for(i in 1:length(list)){

  path <- list[i]
  batch <- batch[i]
  setwd(paste0(path))

  L <- list.files(".", ".mlma")

  # results tables for each protein (all vs sig)
  for(j in 1:length(L)){
    file <- fread(L[j], header = T)
    file <- as.data.frame(file)
    name <- as.character(L[j])
    name <- gsub("_.*", "", name)
    file$SeqId <- name
    write.csv(file, paste0(path, name, "_results_table.csv"), row.names = F)
    sub <- file[-which(file$p > 0.000000036),]
    write.csv(sub, paste0(path, name, "_results_sig.csv"), row.names = F)
    write.csv(sub, paste0(sig_hits, batch, "_", name, "_results_sig.csv"), row.names = F)
  }
}



library(data.table)

list <- c("/Cluster_Filespace/Marioni_Group/Danni/Stradl_markers/EWAS_4000/EWAS_080821_full/batches/batch11/")

sig_hits <- "/Cluster_Filespace/Marioni_Group/Danni/Stradl_markers/EWAS_4000/EWAS_080821_full/batches/sig_hits/"

batch <- c("b11")

for(i in 1:length(list)){

  path <- list[i]
  batch <- batch[i]
  setwd(paste0(path))

  L <- list.files(".", ".mlma")

  # results tables for each protein (all vs sig)
  for(j in 1:length(L)){
    file <- fread(L[j], header = T)
    file <- as.data.frame(file)
    name <- as.character(L[j])
    name <- gsub("_.*", "", name)
    file$SeqId <- name
    write.csv(file, paste0(path, name, "_results_table.csv"), row.names = F)
    sub <- file[-which(file$p > 0.000000036),]
    write.csv(sub, paste0(path, name, "_results_sig.csv"), row.names = F)
    write.csv(sub, paste0(sig_hits, batch, "_", name, "_results_sig.csv"), row.names = F)
  }
}



library(data.table)

list <- c("/Cluster_Filespace/Marioni_Group/Danni/Stradl_markers/EWAS_4000/EWAS_080821_full/batches/batch12/")

sig_hits <- "/Cluster_Filespace/Marioni_Group/Danni/Stradl_markers/EWAS_4000/EWAS_080821_full/batches/sig_hits/"

batch <- c("b12")

for(i in 1:length(list)){

  path <- list[i]
  batch <- batch[i]
  setwd(paste0(path))

  L <- list.files(".", ".mlma")

  # results tables for each protein (all vs sig)
  for(j in 1:length(L)){
    file <- fread(L[j], header = T)
    file <- as.data.frame(file)
    name <- as.character(L[j])
    name <- gsub("_.*", "", name)
    file$SeqId <- name
    write.csv(file, paste0(path, name, "_results_table.csv"), row.names = F)
    sub <- file[-which(file$p > 0.000000036),]
    write.csv(sub, paste0(path, name, "_results_sig.csv"), row.names = F)
    write.csv(sub, paste0(sig_hits, batch, "_", name, "_results_sig.csv"), row.names = F)
  }
}


# Collate top hits for proteins 

path <- "/Cluster_Filespace/Marioni_Group/Danni/Stradl_markers/EWAS_4000/EWAS_080821_full/batches/sig_hits/"

setwd(path)

L <- list.files(".", ".csv")
L # 4230 converged 

files <- lapply(L, read.csv)
names <- as.character(L)
batch <- gsub("_.*", "", names)
marker <- gsub("_results.*", "", names)
marker <- gsub(".*_", "", marker)
names(files) <- marker
osca <- do.call(rbind, files)
osca <- osca[c(9,1:8)]

# > dim(osca)
# [1] 7090 updated to 7247       - number of pQTL adjusted pQTMs 

length(unique(osca$SeqId))

# [1] 355 proteins with signals - updated to 347 proteins with signals

length(files) # 4230

# Read in pheno file again
pheno2 <- read.csv("/Cluster_Filespace/Marioni_Group/Danni/Stradl_markers/EWAS_4000/prep/Phenotype_file_778_eGFR_inlcuded_pQTLs_regressed.csv", check.names = F)
pheno2 <- pheno2[-2]

list <- names(files)
list2 <- pheno2[-which(colnames(pheno2) %in% list)]

    # id    15509-2    15584-9    19141-22    4407-10     6402-8

# NAGLU, CFHR2, DAP, MST1, PILRA

 
# annotations file add in
library(tidyverse) 
library(readxl)
anno <- read_excel("/Cluster_Filespace/Marioni_Group/Danni/Stradl_markers/EWAS_4000/heritability_050321/outputs_combined/SOMAscan_Assay_v4_Annotations_version3.3.2_DG.xlsx")
anno <- as.data.frame(anno)
anno <- anno[c(1,18,4,13)]
comb <- left_join(osca, anno, by = "SeqId")

# Save 
write.csv(comb, "/Cluster_Filespace/Marioni_Group/Danni/Stradl_markers/EWAS_4000/EWAS_080821_full/results/no_eGFR_FULL_EWAS_thr_3_6.csv", row.names = F)

# Filter to CpG-protein correction value of adjusted p 0.05/4235/772619 = 1.528098e-11

# > 0.05 / 4235 / 772619
# [1] 1.528098e-11

# 0.00000000001528098

comb2_filt2 <- comb[which(comb$p < 0.00000000001528098),]

# Save 
write.csv(comb2_filt2, "/Cluster_Filespace/Marioni_Group/Danni/Stradl_markers/EWAS_4000/EWAS_080821_full/results/no_eGFR_FULL_EWAS_thr_bonfer.csv", row.names = F)

# I will use the latter as the correction threshold for now 

dim(comb2_filt2) # 2797  - at cpg/protein adjusted significance - updated to 2895 

# Order and save an ordered version
 comb2 <- comb2_filt2[order(comb2_filt2$p),]

 write.csv(comb2, "/Cluster_Filespace/Marioni_Group/Danni/Stradl_markers/EWAS_4000/EWAS_080821_full/results/no_eGFR_FULL_EWAS_thr_bonfer_ordered.csv", row.names = F)

dim(comb2_filt2) # 2895 



# Compare to see which associations are different between the models
full <- read.csv("/Cluster_Filespace/Marioni_Group/Danni/Stradl_markers/EWAS_4000/EWAS_240521_results/pQTL_WBC_covs_EWAS_thr_bonfer.csv")
full$assoc <- paste(full[,3], full[,1], full[,10], sep = "_")

test <- comb2_filt2
test$assoc <- paste(test[,3], test[,1], test[,10], sep = "_")

test2 <- test[-which(test$assoc %in% full$assoc),]
dim(test2) # 136 added in new models 


test3 <- full[-which(full$assoc %in% test$assoc),]
dim(test3) # 38 lost from previous results with eGFR 



# Get effect size range for the full results 

betas <- comb2[order(comb2$b),]
# -2.64 (0.29) and p = 3.43080e-19 for PRG3 and cg16899419 

comb2[2893:2895,]
# 2.62 (0.19) and p = 6.66156e-37 for cg12415337 and MDGA1 

###################################

### WORK OUT WHICH FULL PROTEINS DIDNT CONVERGE 

# Read in protein file which has GS id and stradl id, then all 4,235 proteins that have been preprocessed
pheno1 <- read.csv("/Cluster_Filespace/Marioni_Group/Danni/Stradl_markers/EWAS_4000/prep/proteins_4235_778_eGFR_included.csv", check.names = F)
pheno2 <- read.csv("/Cluster_Filespace/Marioni_Group/Danni/Stradl_markers/EWAS_4000/prep/Phenotype_file_778_eGFR_removed.csv", check.names = F)
names(pheno2)[1] <- "ID"
pheno2 <- pheno2[-2]
colnames(pheno2) <- colnames(pheno1)


path <- "/Cluster_Filespace/Marioni_Group/Danni/Stradl_markers/EWAS_4000/EWAS_080821_full/batches/sig_hits/"

setwd(path)

L <- list.files(".", ".csv")
L # 4226 converged 

files <- lapply(L, read.csv)
names <- as.character(L)
batch <- gsub("_.*", "", names)
marker <- gsub("_results.*", "", names)
marker <- gsub(".*_", "", marker)
names(files) <- marker
osca <- do.call(rbind, files)
osca <- osca[c(9,1:8)]

# load pheno2 above and do this to work out difference 
list <- names(files)
list2 <- pheno2[-which(colnames(pheno2) %in% list)]

# Plots - of proteins look okay 
proteins <- colnames(list2)[1:8]
proteins




# NON CONVERGENCE MODELS BASIC: 
# [1] "ID"       "11187-11" "15509-2"  "15584-9"  "19141-22" "4407-10"  "6364-7"
# [8] "6402-8"

### PROTEINS THAT DIDNT CONVERGE IN MIDDLE MODEL:
# [1] "ID"       "15509-2"  "15584-9"  "19141-22" "4407-10"  "6402-8"   NA
# [8] NA

### PROTEINS THAT DIDNT CONVERGE IN FULL MODEL:
# [1] "ID"       "15509-2"  "15584-9"  "19141-22" "4407-10"  "6402-8"   NA
# [8] NA


    # id    15509-2    15584-9    19141-22    4407-10     6402-8

# NAGLU, CFHR2, DAP, MST1, PILRA


########################################################################################################

### Subset to nested model structure across models and format tables for suppl reporting 

########################################################################################################

# FIRST FORMAT TABLES 

# Basic: /Cluster_Filespace/Marioni_Group/Danni/Stradl_markers/EWAS_4000/EWAS_240521_results/no_eGFR_basic_EWAS_thr_bonfer.csv
# Middle: /Cluster_Filespace/Marioni_Group/Danni/Stradl_markers/EWAS_4000/EWAS_240521_results/no_eGFR_MIDDLE_EWAS_thr_bonfer.csv
# Full: /Cluster_Filespace/Marioni_Group/Danni/Stradl_markers/EWAS_4000/EWAS_080821_full/results/no_eGFR_FULL_EWAS_thr_bonfer.csv

# Load corrected basic model 
base <- read.csv("/Cluster_Filespace/Marioni_Group/Danni/Stradl_markers/EWAS_4000/EWAS_240521_results/no_eGFR_basic_EWAS_thr_bonfer.csv")
base <- base[c(3,2,5,4,6:9,1,10,11,12)]
names(base) <- c("CpG", "Chromsome of CpG", "Gene of CpG", "CpG position", "Orientation",
  "Beta", "SE", "P", "SeqId", "Gene of protein", "UniProt", "UniProt Full Name") # 133617 remain significant 
base <- base[order(base$P),]
write.csv(base, "/Cluster_Filespace/Marioni_Group/Danni/Stradl_markers/EWAS_4000/EWAS_240521_results/formatted_tables/no_eGFR_basic_131691.csv", row.names = F)

# Load WBC SNP corrected model 
WBC <- read.csv("/Cluster_Filespace/Marioni_Group/Danni/Stradl_markers/EWAS_4000/EWAS_240521_results/no_eGFR_MIDDLE_EWAS_thr_bonfer.csv")
WBC <- WBC[c(3,2,5,4,6:9,1,10,11,12)]
names(WBC) <- c("CpG", "Chromsome of CpG", "Gene of CpG", "CpG position", "Orientation",
  "Beta", "SE", "P", "SeqId", "Gene of protein", "UniProt", "UniProt Full Name") # 2908 remain significant 
WBC <- WBC[order(WBC$P),]
write.csv(WBC, "/Cluster_Filespace/Marioni_Group/Danni/Stradl_markers/EWAS_4000/EWAS_240521_results/formatted_tables/no_eGFR_middle_WBC_pQTL_2974.csv", row.names = F)

# Load full corrected model 
full <- read.csv("/Cluster_Filespace/Marioni_Group/Danni/Stradl_markers/EWAS_4000/EWAS_080821_full/results/no_eGFR_FULL_EWAS_thr_bonfer.csv")
full <- full[c(3,2,5,4,6:9,1,10,11,12)]
names(full) <- c("CpG", "Chromsome of CpG", "Gene of CpG", "CpG position", "Orientation",
  "Beta", "SE", "P", "SeqId", "Gene of protein", "UniProt", "UniProt Full Name") # 2797 remain significant 
full <- full[order(full$P),]
write.csv(full, "/Cluster_Filespace/Marioni_Group/Danni/Stradl_markers/EWAS_4000/EWAS_240521_results/formatted_tables/no_eGFR_full_covs_2895.csv", row.names = F)

# NEXT LOOK AT WHICH ARE CONSISTENT ACROSS MODELS 

# From base to middle model 
base <- read.csv("/Cluster_Filespace/Marioni_Group/Danni/Stradl_markers/EWAS_4000/EWAS_240521_results/no_eGFR_basic_EWAS_thr_bonfer.csv")
WBC <- read.csv("/Cluster_Filespace/Marioni_Group/Danni/Stradl_markers/EWAS_4000/EWAS_240521_results/no_eGFR_MIDDLE_EWAS_thr_bonfer.csv")
keep = paste(base$SeqId, base$Probe, sep = "_")
WBC$retain <- paste(WBC$SeqId, WBC$Probe, sep = "_")
WBC <- WBC[which(WBC$retain %in% keep),]
WBC$retain <- NULL
write.csv(WBC, "/Cluster_Filespace/Marioni_Group/Danni/Stradl_markers/EWAS_4000/EWAS_240521_results/formatted_tables/retained_basic_to_middle_2564_of_2974_unformatted.csv", row.names = F)
WBC <- WBC[c(3,2,5,4,6:9,1,10,11,12)]
names(WBC) <- c("CpG", "Chromsome of CpG", "Gene of CpG", "CpG position", "Orientation",
  "Beta", "SE", "P", "SeqId", "Gene of protein", "UniProt", "UniProt Full Name") # 2484/2908 remain significant with WBC and pQTLs adjusted for from basic model 
write.csv(WBC, "/Cluster_Filespace/Marioni_Group/Danni/Stradl_markers/EWAS_4000/EWAS_240521_results/formatted_tables/no_eGFR_retained_basic_to_middle_2564_of_2974.csv", row.names = F)

# From middle model (filtered) to the final full model 
WBC <- read.csv("/Cluster_Filespace/Marioni_Group/Danni/Stradl_markers/EWAS_4000/EWAS_240521_results/formatted_tables/retained_basic_to_middle_2564_of_2974_unformatted.csv")
full <- read.csv("/Cluster_Filespace/Marioni_Group/Danni/Stradl_markers/EWAS_4000/EWAS_080821_full/results/no_eGFR_FULL_EWAS_thr_bonfer.csv")
keep = paste(WBC$SeqId, WBC$Probe, sep = "_")
full$retain <- paste(full$SeqId, full$Probe, sep = "_")
full <- full[which(full$retain %in% keep),] 
full$retain <- NULL
full <- full[c(3,2,5,4,6:9,1,10,11,12)]
names(full) <- c("CpG", "Chromsome of CpG", "Gene of CpG", "CpG position", "Orientation",
  "Beta", "SE", "P", "SeqId", "Gene of protein", "UniProt", "UniProt Full Name") # 2357/2484 remain significant with lifestyle covs added 
write.csv(full, "/Cluster_Filespace/Marioni_Group/Danni/Stradl_markers/EWAS_4000/EWAS_240521_results/formatted_tables/retained_middle_to_full_covs_2451_of_2895.csv", row.names = F)

# Look at how many we lose from the final model overall 
pqtl <- read.csv("/Cluster_Filespace/Marioni_Group/Danni/Stradl_markers/EWAS_4000/EWAS_240521_results/formatted_tables/retained_basic_to_middle_2564_of_2974_unformatted.csv")
filt <- read.csv("/Cluster_Filespace/Marioni_Group/Danni/Stradl_markers/EWAS_4000/EWAS_240521_results/formatted_tables/retained_middle_to_full_covs_2451_of_2895.csv")

pqtl$retain = paste(pqtl$SeqId, pqtl$Probe, sep = "_")
filt$retain <- paste(filt$SeqId, filt$CpG, sep = "_")

diff <- pqtl[which(!pqtl$retain %in% filt$retain),]
names(diff)[10] <- "gene"
unique(diff$gene)

diff <- pqtl[which(pqtl$retain %in% filt$retain),] # 2451

# # Lets look at the 440 to see if they are mostly PAPPA or other proteins 
# new <- pqtl[-which(pqtl$retain %in% filt$retain),]
# write.csv(new, "/Cluster_Filespace/Marioni_Group/Danni/Stradl_markers/EWAS_4000/EWAS_240521_results/formatted_tables/the_440_lost_from_final_model.csv", row.names = F)

# # most are either PAPPA or PRG3 - lets remove them and see how many were other protein signals 

# new <- new[-which(new$Gene.Name.Name %in% "PAPPA"),]
# new <- new[-which(new$Gene.Name.Name %in% "PRG3"),] 

# dim(new) # 80 - there are 80/440 that are non-PAPPA and non-PRG3 pQTMs 

# dim(unique(new[10])) # - 56 unique proteins in the group of 80

# write.csv(new, "/Cluster_Filespace/Marioni_Group/Danni/Stradl_markers/EWAS_4000/EWAS_240521_results/formatted_tables/the_80_of_440_lost_from_final_model_that_arent_PAPPA_or_PRG3.csv", row.names = F)

##############################################################################

### NOW LOOK AT WHETHER WE REPLICATE ZAGHLOOL ET AL 98 pQTM RESULTS VS FULL 

##############################################################################

# Read in zaghlool et al results file with 98 pQTMs and key stats 
library(readxl)
library(tidyverse)
zag <- read_excel("/Cluster_Filespace/Marioni_Group/Danni/Stradl_markers/EWAS_4000/EWAS_240521_results/formatted_tables/Zaghlool_41467_2019_13831_MOESM5_ESM_EDITED.xlsx")
zag <- as.data.frame(zag)
zag$proteinID <- sub("\\_.", "", zag$proteinID) # get seqIds for matching 
list <- zag$proteinID
zag$sig <- paste(zag$proteinID, zag$cpg, sep = "_")

# Work out how many comparable proteins there are accross the 2 studies that could in theory be replicated (i.e. which of the 98 pQTMs did we test for in our EWAS)
stradl <- read.csv("/Cluster_Filespace/Marioni_Group/Danni/Stradl_markers/EWAS_4000/prep/Phenotype_file_778_eGFR_inlcuded_pQTLs_regressed.csv", check.names = F)
length(which(zag$proteinID %in% colnames(stradl))) # 94 matched 

zag_not <- zag[which(!zag$proteinID %in% colnames(stradl)),]

# 4 which were not comparable across the 2 studies (involving 3 proteins)
#    order   label proteinID        cpg            p       beta         se   N
# 29    29   NLRC5   5099-14 cg07839457 1.288322e-17 -0.2742358 0.03146097 935
# 70    70   NLRC5   5099-14 cg08159663 9.221454e-13 -0.2458608 0.03386822 800
# 83    83   PRTN3   3514-49 cg27535410 1.205885e-11 -0.2187312 0.03186051 938
# 94    94 SIGLEC5    5125-6 cg09488502 4.893690e-11  0.2202263 0.03307355 869
#                   sig
# 29 5099-14_cg07839457
# 70 5099-14_cg08159663
# 83 3514-49_cg27535410
# 94  5125-6_cg09488502

# make sure only 94 comparable included 
zag <- zag[which(zag$proteinID %in% colnames(stradl)),]

### make sure all cpgs in the zaghlool et al list are present n our meth dataset used 

first read the meth data in as above 
## Read in methylation file used for BayesR and transpose so CpGs are columns and IDs are rows
library(data.table) 
w2 = readRDS("/Cluster_Filespace/Marioni_Group/Rob/Somalogic/wave2-STRADL-mvals.rds")
w3 = readRDS("/Cluster_Filespace/Marioni_Group/Rob/Somalogic/wave3-STRADL-mvals.rds")
w2 = w2[which(row.names(w2) %in% row.names(w3)),]
w3 = w3[which(row.names(w3) %in% row.names(w2)),]

## Combine w2 and w3 
meth = cbind(w2,w3)

## Add in GS ids instead of DNAm_ids 
stradl_ids <- read.table("/Cluster_Filespace/Marioni_Group/Rob/Somalogic/STRADL_DNAm_target_REM_17April2020.txt",header =T)
id.778 <- read.csv("/Cluster_Filespace/Marioni_Group/Rob/Somalogic/IDs_778.csv")

## Subset to the people in the ID file 
stradl_ids <- stradl_ids[which(stradl_ids$GS_id %in% id.778$x),]
ids = stradl_ids$DNAm_id
meth <- meth[,match(ids, colnames(meth))]
table(stradl_ids$DNAm_id == colnames(meth))
colnames(meth) <- stradl_ids$GS_id

# next look for common cpgs across both studies for the 98 pQTMs
meth <- meth 
dim(meth) #  772619    778
length(which(zag$cpg %in% rownames(meth))) # only 81 CpGs are common to both studies 

# lets look at which are not included 

not_cpg <- zag[which(!zag$cpg %in% rownames(meth)),]


# > not_cpg
#    order label proteinID        cpg            p       beta         se   N
# 7      7 ICAM5   5124-69 cg10604476 6.092957e-25  0.3556380 0.03334739 812
# 12    12 PAPPA   4148-49 cg07562835 6.053805e-24 -0.3245133 0.03126409 920
# 19    19 PAPPA   4148-49 cg22805603 2.312486e-20  0.2952428 0.03119564 938
# 26    26 PAPPA   4148-49 cg18463686 3.845588e-18  0.2802216 0.03160582 923
# 38    38 PAPPA   4148-49 cg07708453 3.103772e-16  0.2621269 0.03150946 938
# 45    45 PAPPA   4148-49 cg03149567 2.478359e-15 -0.2553335 0.03170958 929
# 49    49 PAPPA   4148-49 cg06756385 1.092719e-14 -0.2484275 0.03162756 938
# 50    50 NLRC5   3311-27 cg08159663 1.441578e-14 -0.2898182 0.03682426 664
# 69    69 NLRC5   3485-28 cg00218406 3.968618e-13 -0.2340531 0.03179146 934
# 77    77 NLRC5   3292-75 cg08159663 3.382490e-12 -0.2394780 0.03387316 800
# 78    78 PAPPA   4148-49 cg11456013 3.891687e-12 -0.2261974 0.03215273 918
# 85    85    C4   4481-34 cg13028630 1.375174e-11 -0.2457877 0.03577153 724
# 96    96 PAPPA   4148-49 cg26922697 7.399367e-11 -0.2103716 0.03193007 937
#                   sig
# 7  5124-69_cg10604476
# 12 4148-49_cg07562835
# 19 4148-49_cg22805603
# 26 4148-49_cg18463686
# 38 4148-49_cg07708453
# 45 4148-49_cg03149567
# 49 4148-49_cg06756385
# 50 3311-27_cg08159663
# 69 3485-28_cg00218406
# 77 3292-75_cg08159663
# 78 4148-49_cg11456013
# 85 4481-34_cg13028630
# 96 4148-49_cg26922697





# We need to therefore subset to just those associations which are comaprable (i.e proteins and CpGs)
# 94 pQTMs currently in zag file - now remove CpGs that arent comparable 
zag <- zag[which(zag$cpg %in% rownames(meth)),]
dim(zag) # 81  9
# so a possible 81 pQTMs are comparable across our studies 


# Read in results files 
res <- read.csv("/Cluster_Filespace/Marioni_Group/Danni/Stradl_markers/EWAS_4000/EWAS_240521_results/formatted_tables/no_eGFR_full_covs_2895.csv")
res <- res[which(res$SeqId %in% list),]
res$index <- paste(res$SeqId, res$CpG, sep = "_")

length(which(zag$sig %in% res$index)) # 31 present in our set (full)

# save the subset of 31 out for colour coding results 
save <- zag[which(zag$sig %in% res$index),]

# What about at EWAS levels of significance 
res <- read.csv("/Cluster_Filespace/Marioni_Group/Danni/Stradl_markers/EWAS_4000/EWAS_080821_full/results/no_eGFR_FULL_EWAS_thr_3_6.csv")
res <- res[which(res$SeqId %in% list),]
res$index <- paste(res$SeqId, res$Probe, sep = "_")

length(which(zag$sig %in% res$index)) # 47 replicate at lower trheshold of significance
rep <- zag[which(zag$sig %in% res$index),]

# Make a replication table 
table <- zag[c(2:7,9)]
names(res)[13] <- "sig"

table <- left_join(zag, res, by= "sig")

# save a copy of this table 
table <- table[c(3,2,4,6,7,5,16,17,18,21)]
colnames(table) <- c("SeqId", "Gene of Protein", "CpG", "Zaghlool Beta", "Zaghlool SE", "Zaghlool P", "Beta", "SE", "P", "UniProt Full Name")
write.csv(table, "/Cluster_Filespace/Marioni_Group/Danni/Stradl_markers/EWAS_4000/EWAS_180821_results/replication_no_eGFR/no_eGFR_replication_zaghlool_missing_data_170821.csv", row.names = F)

### Now we will need to find the effect stats for the associations which do not pass significance for the proteins in the table remaining 

# ill need to fish out the EWAS results for these proteins now 

# 3216-2 - 1 association - this protein is in batch 7 
p1 <- read.csv("/Cluster_Filespace/Marioni_Group/Danni/Stradl_markers/EWAS_4000/EWAS_080821_full/batches/batch7/3216-2_results_table.csv")

# 3485-28 - 2 associations - batch 8
p2 <- read.csv("/Cluster_Filespace/Marioni_Group/Danni/Stradl_markers/EWAS_4000/EWAS_080821_full/batches/batch8/3485-28_results_table.csv")

# 4990-87 - 1 association - batch 8 
p3 <- read.csv("/Cluster_Filespace/Marioni_Group/Danni/Stradl_markers/EWAS_4000/EWAS_080821_full/batches/batch8/4990-87_results_table.csv")

# 4148-49 - rest (32) are this protein - batch 8 
p4 <- read.csv("/Cluster_Filespace/Marioni_Group/Danni/Stradl_markers/EWAS_4000/EWAS_080821_full/batches/batch8/4148-49_results_table.csv")

# so of the 36, there were 32 for PAPPA and 4148-49 that didnt replicate 

# Im going to join the tables, name protein_cpg column and see if we can match in the stats for our replication table 

p <- rbind(p1,p2)
p <- rbind(p,p3)
p <- rbind(p,p4)

# get joiner column
table$assoc <- paste(table$SeqId, table$CpG, sep = "_")
p$assoc <- paste(p$SeqId, p$Probe, sep = "_")

# name appropriately 
p2 <- p[c(10,6,7,8)]
names(p2) <- c("assoc", "b_non", "SE_non", "p_non")

# Can we find the associations of interest 
names(table)[10] <- "Uni"
sub <- table[which(table$Uni %in% NA),] # 34 associations to fill in 

length(which(sub$assoc %in% p2$assoc)) # 34 present 

# try to merge in

merge <- left_join(table, p2, by = "assoc")

# write out this file and examine to make sure we have results for every instance of the 81 comparable 
write.csv(merge, "/Cluster_Filespace/Marioni_Group/Danni/Stradl_markers/EWAS_4000/EWAS_180821_results/replication_no_eGFR/replication_zaghlool_missing_data_added_180821.csv", row.names = F)

# this will be our replication suppl table - now do the suppl figure to illustrate it too 

# to do this, i'll read back in the file that i edited to collate the missing vs non missing data added in 
merge <- read_excel("/Cluster_Filespace/Marioni_Group/Danni/Stradl_markers/EWAS_4000/EWAS_240521_results/replication_zaghlool_missing_data_added_080721.xlsx")
merge <- as.data.frame(merge)

# ### Plot the replication of associations

# names(merge)[4] <- "zb"
# names(merge)[7] <- "ob"

# # save as variable for patchworking 
# plot1 <- ggplot(merge, aes(zb, ob), scale="globalminmax") +
#        geom_vline(xintercept = 0, linetype = 1) +
#        geom_hline(yintercept = 0, linetype = 1) +
#        geom_point() +
#        theme_minimal() +
#       xlab("Beta effect Zaghlool et al") +
#       ylab("Beta effect") +
#        theme(plot.title = element_text(size = 14)) +
#    geom_abline(slope=1, intercept=0, linetype = 2) + xlim(-3,3) + ylim(-3,3)

#    +
#   geom_point(data=merge[merge$Rep_both =="1",], shape=21, fill="white", size=1)+
#   geom_point(data=merge[merge$Rep_both !="1",], colour='red') 



##############################################################################

### NOW LOOK AT LAMBDA VALUE CALCULATION FOR THE FULL MODELS   

##############################################################################

## OSCA Lambdas 

library(data.table)

cpgs <- read.csv("/Cluster_Filespace/Marioni_Group/Danni/Stradl_markers/EWAS_4000/EWAS_240521_results/formatted_tables/no_eGFR_full_covs_2895.csv")
length <- dim(cpgs)[1]
list = unique(cpgs$SeqId) # 153 proteins to get lambdas for 
names(cpgs)[10] <- "Somamer"

output <- matrix(nrow= length, ncol = 3)
output <- as.data.frame(output)
names(output) <- c("Protein", "Gene of Protein", "Lambda")

# Run for each batch to get lambdas and join at the end 

for(i in 1:length(list)){ 
  tryCatch({ 
    ewas = fread(paste0("/Cluster_Filespace/Marioni_Group/Danni/Stradl_markers/EWAS_4000/EWAS_080821_full/batches/batch1/", list[[i]], "_MOA_778_with_ORM.mlma"))
    ewas = as.data.frame(ewas)

    gene = cpgs[cpgs$SeqId %in% list[[i]],"Somamer"]
    gene = unique(gene)

    ## Calculate Lambda 
    # For p-values, calculate chi-squared statistic
    chisq <- qchisq(1-ewas$p,1)

    ## Calculate lambda gc (λgc)
    lambda <- median(chisq)/qchisq(0.5,1)

    output[i, 1] <- list[[i]]
    output[i, 2] <- gene
    output[i, 3] <- lambda 
  }, error = function(e) cat("skipped"))
} 

write.csv(output, "/Cluster_Filespace/Marioni_Group/Danni/Stradl_markers/EWAS_4000/EWAS_180821_results/lambdas_no_eGFR/batch1_inflation.csv", row.names = F)





for(i in 1:length(list)){ 
  tryCatch({ 
    ewas = fread(paste0("/Cluster_Filespace/Marioni_Group/Danni/Stradl_markers/EWAS_4000/EWAS_080821_full/batches/batch2/", list[[i]], "_MOA_778_with_ORM.mlma"))
    ewas = as.data.frame(ewas)

    gene = cpgs[cpgs$SeqId %in% list[[i]],"Somamer"]
    gene = unique(gene)

    ## Calculate Lambda 
    # For p-values, calculate chi-squared statistic
    chisq <- qchisq(1-ewas$p,1)

    ## Calculate lambda gc (λgc)
    lambda <- median(chisq)/qchisq(0.5,1)

    output[i, 1] <- list[[i]]
    output[i, 2] <- gene
    output[i, 3] <- lambda 
  }, error = function(e) cat("skipped"))
} 

write.csv(output, "/Cluster_Filespace/Marioni_Group/Danni/Stradl_markers/EWAS_4000/EWAS_180821_results/lambdas_no_eGFR/batch2_inflation.csv", row.names = F)




for(i in 1:length(list)){ 
  tryCatch({ 
    ewas = fread(paste0("/Cluster_Filespace/Marioni_Group/Danni/Stradl_markers/EWAS_4000/EWAS_080821_full/batches/batch3/", list[[i]], "_MOA_778_with_ORM.mlma"))
    ewas = as.data.frame(ewas)

    gene = cpgs[cpgs$SeqId %in% list[[i]],"Somamer"]
    gene = unique(gene)

    ## Calculate Lambda 
    # For p-values, calculate chi-squared statistic
    chisq <- qchisq(1-ewas$p,1)

    ## Calculate lambda gc (λgc)
    lambda <- median(chisq)/qchisq(0.5,1)

    output[i, 1] <- list[[i]]
    output[i, 2] <- gene
    output[i, 3] <- lambda 
  }, error = function(e) cat("skipped"))
} 

write.csv(output, "/Cluster_Filespace/Marioni_Group/Danni/Stradl_markers/EWAS_4000/EWAS_180821_results/lambdas_no_eGFR/batch3_inflation.csv", row.names = F)



for(i in 1:length(list)){ 
  tryCatch({ 
    ewas = fread(paste0("/Cluster_Filespace/Marioni_Group/Danni/Stradl_markers/EWAS_4000/EWAS_080821_full/batches/batch4/", list[[i]], "_MOA_778_with_ORM.mlma"))
    ewas = as.data.frame(ewas)

    gene = cpgs[cpgs$SeqId %in% list[[i]],"Somamer"]
    gene = unique(gene)

    ## Calculate Lambda 
    # For p-values, calculate chi-squared statistic
    chisq <- qchisq(1-ewas$p,1)

    ## Calculate lambda gc (λgc)
    lambda <- median(chisq)/qchisq(0.5,1)

    output[i, 1] <- list[[i]]
    output[i, 2] <- gene
    output[i, 3] <- lambda 
  }, error = function(e) cat("skipped"))
} 

write.csv(output, "/Cluster_Filespace/Marioni_Group/Danni/Stradl_markers/EWAS_4000/EWAS_180821_results/lambdas_no_eGFR/batch4_inflation.csv", row.names = F)



for(i in 1:length(list)){ 
  tryCatch({ 
    ewas = fread(paste0("/Cluster_Filespace/Marioni_Group/Danni/Stradl_markers/EWAS_4000/EWAS_080821_full/batches/batch5/", list[[i]], "_MOA_778_with_ORM.mlma"))
    ewas = as.data.frame(ewas)

    gene = cpgs[cpgs$SeqId %in% list[[i]],"Somamer"]
    gene = unique(gene)

    ## Calculate Lambda 
    # For p-values, calculate chi-squared statistic
    chisq <- qchisq(1-ewas$p,1)

    ## Calculate lambda gc (λgc)
    lambda <- median(chisq)/qchisq(0.5,1)

    output[i, 1] <- list[[i]]
    output[i, 2] <- gene
    output[i, 3] <- lambda 
  }, error = function(e) cat("skipped"))
} 

write.csv(output, "/Cluster_Filespace/Marioni_Group/Danni/Stradl_markers/EWAS_4000/EWAS_180821_results/lambdas_no_eGFR/batch5_inflation.csv", row.names = F)



for(i in 1:length(list)){ 
  tryCatch({ 
    ewas = fread(paste0("/Cluster_Filespace/Marioni_Group/Danni/Stradl_markers/EWAS_4000/EWAS_080821_full/batches/batch6/", list[[i]], "_MOA_778_with_ORM.mlma"))
    ewas = as.data.frame(ewas)

    gene = cpgs[cpgs$SeqId %in% list[[i]],"Somamer"]
    gene = unique(gene)

    ## Calculate Lambda 
    # For p-values, calculate chi-squared statistic
    chisq <- qchisq(1-ewas$p,1)

    ## Calculate lambda gc (λgc)
    lambda <- median(chisq)/qchisq(0.5,1)

    output[i, 1] <- list[[i]]
    output[i, 2] <- gene
    output[i, 3] <- lambda 
  }, error = function(e) cat("skipped"))
} 

write.csv(output, "/Cluster_Filespace/Marioni_Group/Danni/Stradl_markers/EWAS_4000/EWAS_180821_results/lambdas_no_eGFR/batch6_inflation.csv", row.names = F)



for(i in 1:length(list)){ 
  tryCatch({ 
    ewas = fread(paste0("/Cluster_Filespace/Marioni_Group/Danni/Stradl_markers/EWAS_4000/EWAS_080821_full/batches/batch7/", list[[i]], "_MOA_778_with_ORM.mlma"))
    ewas = as.data.frame(ewas)

    gene = cpgs[cpgs$SeqId %in% list[[i]],"Somamer"]
    gene = unique(gene)

    ## Calculate Lambda 
    # For p-values, calculate chi-squared statistic
    chisq <- qchisq(1-ewas$p,1)

    ## Calculate lambda gc (λgc)
    lambda <- median(chisq)/qchisq(0.5,1)

    output[i, 1] <- list[[i]]
    output[i, 2] <- gene
    output[i, 3] <- lambda 
  }, error = function(e) cat("skipped"))
} 

write.csv(output, "/Cluster_Filespace/Marioni_Group/Danni/Stradl_markers/EWAS_4000/EWAS_180821_results/lambdas_no_eGFR/batch7_inflation.csv", row.names = F)



for(i in 1:length(list)){ 
  tryCatch({ 
    ewas = fread(paste0("/Cluster_Filespace/Marioni_Group/Danni/Stradl_markers/EWAS_4000/EWAS_080821_full/batches/batch8/", list[[i]], "_MOA_778_with_ORM.mlma"))
    ewas = as.data.frame(ewas)

    gene = cpgs[cpgs$SeqId %in% list[[i]],"Somamer"]
    gene = unique(gene)

    ## Calculate Lambda 
    # For p-values, calculate chi-squared statistic
    chisq <- qchisq(1-ewas$p,1)

    ## Calculate lambda gc (λgc)
    lambda <- median(chisq)/qchisq(0.5,1)

    output[i, 1] <- list[[i]]
    output[i, 2] <- gene
    output[i, 3] <- lambda 
  }, error = function(e) cat("skipped"))
} 

write.csv(output, "/Cluster_Filespace/Marioni_Group/Danni/Stradl_markers/EWAS_4000/EWAS_180821_results/lambdas_no_eGFR/batch8_inflation.csv", row.names = F)



for(i in 1:length(list)){ 
  tryCatch({ 
    ewas = fread(paste0("/Cluster_Filespace/Marioni_Group/Danni/Stradl_markers/EWAS_4000/EWAS_080821_full/batches/batch9/", list[[i]], "_MOA_778_with_ORM.mlma"))
    ewas = as.data.frame(ewas)

    gene = cpgs[cpgs$SeqId %in% list[[i]],"Somamer"]
    gene = unique(gene)

    ## Calculate Lambda 
    # For p-values, calculate chi-squared statistic
    chisq <- qchisq(1-ewas$p,1)

    ## Calculate lambda gc (λgc)
    lambda <- median(chisq)/qchisq(0.5,1)

    output[i, 1] <- list[[i]]
    output[i, 2] <- gene
    output[i, 3] <- lambda 
  }, error = function(e) cat("skipped"))
} 

write.csv(output, "/Cluster_Filespace/Marioni_Group/Danni/Stradl_markers/EWAS_4000/EWAS_180821_results/lambdas_no_eGFR/batch9_inflation.csv", row.names = F)



for(i in 1:length(list)){ 
  tryCatch({ 
    ewas = fread(paste0("/Cluster_Filespace/Marioni_Group/Danni/Stradl_markers/EWAS_4000/EWAS_080821_full/batches/batch10/", list[[i]], "_MOA_778_with_ORM.mlma"))
    ewas = as.data.frame(ewas)

    gene = cpgs[cpgs$SeqId %in% list[[i]],"Somamer"]
    gene = unique(gene)

    ## Calculate Lambda 
    # For p-values, calculate chi-squared statistic
    chisq <- qchisq(1-ewas$p,1)

    ## Calculate lambda gc (λgc)
    lambda <- median(chisq)/qchisq(0.5,1)

    output[i, 1] <- list[[i]]
    output[i, 2] <- gene
    output[i, 3] <- lambda 
  }, error = function(e) cat("skipped"))
} 

write.csv(output, "/Cluster_Filespace/Marioni_Group/Danni/Stradl_markers/EWAS_4000/EWAS_180821_results/lambdas_no_eGFR/batch10_inflation.csv", row.names = F)



for(i in 1:length(list)){ 
  tryCatch({ 
    ewas = fread(paste0("/Cluster_Filespace/Marioni_Group/Danni/Stradl_markers/EWAS_4000/EWAS_080821_full/batches/batch11/", list[[i]], "_MOA_778_with_ORM.mlma"))
    ewas = as.data.frame(ewas)

    gene = cpgs[cpgs$SeqId %in% list[[i]],"Somamer"]
    gene = unique(gene)

    ## Calculate Lambda 
    # For p-values, calculate chi-squared statistic
    chisq <- qchisq(1-ewas$p,1)

    ## Calculate lambda gc (λgc)
    lambda <- median(chisq)/qchisq(0.5,1)

    output[i, 1] <- list[[i]]
    output[i, 2] <- gene
    output[i, 3] <- lambda 
  }, error = function(e) cat("skipped"))
} 

write.csv(output, "/Cluster_Filespace/Marioni_Group/Danni/Stradl_markers/EWAS_4000/EWAS_180821_results/lambdas_no_eGFR/batch11_inflation.csv", row.names = F)



for(i in 1:length(list)){ 
  tryCatch({ 
    ewas = fread(paste0("/Cluster_Filespace/Marioni_Group/Danni/Stradl_markers/EWAS_4000/EWAS_080821_full/batches/batch12/", list[[i]], "_MOA_778_with_ORM.mlma"))
    ewas = as.data.frame(ewas)

    gene = cpgs[cpgs$SeqId %in% list[[i]],"Somamer"]
    gene = unique(gene)

    ## Calculate Lambda 
    # For p-values, calculate chi-squared statistic
    chisq <- qchisq(1-ewas$p,1)

    ## Calculate lambda gc (λgc)
    lambda <- median(chisq)/qchisq(0.5,1)

    output[i, 1] <- list[[i]]
    output[i, 2] <- gene
    output[i, 3] <- lambda 
  }, error = function(e) cat("skipped"))
} 

write.csv(output, "/Cluster_Filespace/Marioni_Group/Danni/Stradl_markers/EWAS_4000/EWAS_180821_results/lambdas_no_eGFR/batch12_inflation.csv", row.names = F)


## Collate the lambda values across batches to get the full set 
setwd("/Cluster_Filespace/Marioni_Group/Danni/Stradl_markers/EWAS_4000/EWAS_180821_results/lambdas_no_eGFR/")
L <- list.files("/Cluster_Filespace/Marioni_Group/Danni/Stradl_markers/EWAS_4000/EWAS_180821_results/lambdas_no_eGFR/", ".")

files <- lapply(L, read.csv)
files <- na.omit(files)
osca <- do.call(rbind, files)
osca <- na.omit(osca)
names(osca) <- c("Protein SeqId", "Gene of Protein", "Lambda")

# write.csv(osca, "/Cluster_Filespace/Marioni_Group/Danni/Stradl_markers/EWAS_4000/EWAS_180821_results/lambdas_no_eGFR/joint/joint.csv", row.names = F)

osca2 <- osca[which(osca[,1] %in% list),]

unique(osca2[,1])

library(tidyverse)
osca3 <- osca2 %>% group_by("Protein SeqId")

# It seems we are duplicating rows that are identical  -remove duplicates

table(duplicated(osca))

unique <- unique(osca)
dim(unique) # 153 proteins - as expected 

unique <- unique[order(unique$Lambda),]

write.csv(unique, "/Cluster_Filespace/Marioni_Group/Danni/Stradl_markers/EWAS_4000/EWAS_180821_results/lambdas_no_eGFR/joint/joint.csv", row.names = F)


##########################################################################################################################

# Notes on the inflation and false positives from rob conversation

# At this bonfer threshold we'd see 10-20 assocs by chance 
# Crude - because there will be other relationships between confounders and how that changes in our sample relative to real world
# Everything with PAPPA and PRG3 - signal across epigenome, nothing to do with that 
# Basically ignore those and focus on other ones 
# PAPPA or PRG3 not real - correlated with each other or cell counts 


# So lambdas are used as genomic inflation factors in GWAS
# Q-Q plot - straight line of p value association stats which follow this stright line then curl up 
# At top - point where most significant SNPs - big tail = loads that are super significant 
# It could be real or fake - if its real people call it polygenicity 
# That can be a problem in itself, as its the sum of many small effects - as you look at each SNP in turn, youll see polygenicity in presence
# of a counfounder, or in presence of something true - lots to tease apart what is happening - population sturcture is a big one - so 
# correlations between people causing multiple SNPs to get  flagged up with stronger assocs stats for SNPS as youre looking at those with similar backgrounds
# Polygfenicity & pop structure are like the real and fake effects - even though polygenicity cant be trusted 
# Because we look at each SNP in turn 

# In EWAS, if we run each CpG in turn, like limma, without accounting for intercorrelations between CpGs, we see probably a worse situation, 
# due to all the confounders like cell types etc. Like we see for PAPPA - super inflated EWAS. Can be due to anything, - rob had CRTAM which has B cell confounding 
# down to 10 people being outliers for B cell methylation. If its really inflated in EWAS its probably not true and youre probs looking at assocation stats due to 
# intercorrelations between CpGs.

# Linear reg suffers from this - turn to other methods that can appropriately account for these - mixed model approaches in OSCA - fit all CpGs as 
# random effects, allowing for different slopes and intercepts - it can control for their intercorrelations much better - because they are randome effects and youre
# modelling them for each cpG site - then youre more likely to capture unknown confounders, as your read out for the unknown confounders is this intercorrelation between CpG sites 
# CpG sites for each test or correlation that you run, so its not important that you knew the confounder, its just important that you know what each has done to the 
# CpG site, one CpG site & its relationship with others - thats why itll reduce intercorrelations and bring down inflation but sometimes its too strong correlation and 
# cant model it appropriately (ie PAPPA) 
# if its 1 - no association signal so kind of a trade off 
# sometimes you get inflation due to strong signals 
# with OSCA we'd expect something lower than linear method as they will be expecting a ranodm effect structure to adjust for inflation.

# You are allowing them to have different slopes and intercepts - intuitive 
# Lambda - spread chisq dist, then you take median of that and divide it by the associations you expect under the null 
# Median of what you see, vs median of what you see under null 

